units/migrate_volume_job.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	// Avoid recreating new host on retry
+	if j.NewHostID == "" {
+		if hasErr := j.startNewHost(ctx); hasErr {
+			return
+		}
+	}
+
+	// Terminate initial host
+	if !j.InitialHostTerminated {
+		ts := utility.RoundPartOfMinute(1).Format(TSFormat)
+		terminateJob := NewSpawnHostTerminationJob(j.initialHost, evergreen.User, ts)
+		if err := j.env.RemoteQueue().Put(ctx, terminateJob); err != nil {
+			j.AddRetryableError(errors.Wrapf(err, "enqueueing initial host termination job '%s'", terminateJob.ID()))
+			return
+		}
+		j.InitialHostTerminated = true
+	}
+}
+
+// stopInitialHost inspects the initial host's status and stops it if the host is still running.
+// It returns a boolean indicating whether the host has stopped and a boolean indicating whether an error was added to the job.
+func (j *volumeMigrationJob) stopInitialHost(ctx context.Context) (bool, bool) {
+	// Stop host if this operation has not been attempted yet
+	if j.SpawnhostStopJobID == "" {
+		ts := utility.RoundPartOfMinute(1).Format(TSFormat)
+		stopJob := NewSpawnhostStopJob(j.initialHost, evergreen.User, ts)
+		err := j.env.RemoteQueue().Put(ctx, stopJob)
+		if err != nil {
+			j.AddRetryableError(err)
+			return false, true
+		}
+		grip.Info(message.Fields{
+			"message": "stopping initial host",
+			"host_id": j.InitialHostID,
+		})
+		j.SpawnhostStopJobID = stopJob.ID()
+		return false, false
+	}
+
+	// Mark host as stopped
+	if j.initialHost.Status == evergreen.HostStopped {
+		grip.Info(message.Fields{
+			"message": "initial host stopped",
+			"host_id": j.InitialHostID,
+		})
+		j.InitialHostStopped = true
+		return true, false
+	}
+
+	// If host is not stopped, verify whether the job is still in progress and error if it completed without stopping
+	stopJob, foundJob := j.env.RemoteQueue().Get(ctx, j.SpawnhostStopJobID)
+	if !foundJob {
+		j.AddRetryableError(errors.Errorf("spawnhost-stop job '%s' could not be found", j.SpawnhostStopJobID))
+		return false, true
+	}
+
+	if stopJob.Status().Completed {
+		j.AddError(errors.Errorf("host '%s' failed to stop", j.SpawnhostStopJobID))
+		return false, true
+	}
+
+	return false, false
+}
+
+// startNewHost attempts to start a new host with the volume attached
+// It returns a boolean indicating if an error was added to the job
+func (j *volumeMigrationJob) startNewHost(ctx context.Context) bool {
+	// Ensure volume has been detached
+	if j.volume.Host == j.InitialHostID {
+		j.AddRetryableError(errors.New("volume still attached to host"))
+		return true
+	}
+
+	j.ModifyOptions.HomeVolumeID = j.VolumeID
+	intentHost, err := cloud.CreateSpawnHost(ctx, j.ModifyOptions, j.env.Settings())
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "creating new intent host"))
+		return true
+	}
+
+	if err := intentHost.Insert(); err != nil {
+		j.AddError(errors.Wrap(err, "inserting new intent host"))
+		return true
+	}
+	grip.Info(message.Fields{
+		"message":  "new intent host created",
+		"attempts": j.RetryInfo().CurrentAttempt,
```

Comment:
Can we add host IDs / job IDs etc to these logs? Otherwise they're not going to be super useful and we should just remove them.